lib/node.js

```diff
@@ -664,9 +664,47 @@ declare module "https" {
 }
 
 declare module "net" {
+
+  declare class Server extends events$EventEmitter {
+    listen(port: number, hostname?: string, backlog?: number, callback?: Function): Server;
+    listen(path: string, callback?: Function): Server;
+    listen(handle: Object, callback?: Function): Server;
+    close(callback?: Function): Server;
+    address(): { port: number; fmaily: string; address: string; };
+    maxConnections: number;
+  }
+
+  declare class Socket extends events$EventEmitter {
+    //TODO
+  }
+
   declare function isIP(input: string): number;
   declare function isIPv4(input: string): boolean;
   declare function isIPv6(input: string): boolean;
+
+
+  declare type connectionListener = (socket: Socket) => any;
+  declare function createServer(
+    options?: {
+      allowHalfOpen: boolean,
```

Comment:
i think these keys should be optional too. `net.createServer({pauseOnConnect: false})` is valid, right?